comp/languagedetection/client/client.go

```diff
@@ -352,15 +352,17 @@ func (c *client) handlePodEvent(podEvent workloadmeta.Event) {
 	for _, c := range append(pod.InitContainers, pod.Containers...) {
 		containerIDs = append(containerIDs, c.ID)
 	}
-	if podEvent.Type == workloadmeta.EventTypeSet {
-		c.retryProcessEventsWithoutPod(containerIDs)
-	}
 
-	for _, cid := range containerIDs {
-		delete(c.processesWithoutPod, cid)
+	switch podEvent.Type {
+	case workloadmeta.EventTypeSet:
+		c.retryProcessEventsWithoutPod(containerIDs)
+	case workloadmeta.EventTypeUnset:
+		delete(c.currentBatch, pod.Name)
+		delete(c.freshlyUpdatedPods, pod.Name)
```

Comment:
We should delete the pod from the `currentBatch` and from `freshlyUpdatedPods` only if we receive an Unset event